docs/libcudacxx/extended_api/utility/static_for.rst

```diff
@@ -0,0 +1,80 @@
+.. _libcudacxx-extended-api-utility-static-for:
+
+``static_for``
+==============
+
+Defined in ``<cuda/utility>`` header.
```

Comment:
Question: Do we want to expose this is a publicly supported API, when there is (likely) a better language construct in C++26, which replaces it? See [expansion statements](http://wg21.link/P1306) (in wording review for C++26)

I am not against adding it, just thinking loudly.

Reply:
Given that we currently do not even support C++23 I would say that a library feature is a good idea

Reply:
Works for me!